Bluetooth: Mesh: Add persistence for schedule register (#4373)
nrfconnect/sdk-nrf: include/bluetooth/mesh/scheduler_srv.rst

Persistent storage
******************

The Scheduler Server stores the following information:

* Any changes to the Schedule Register state

This information is used to restore previously configured register entries when the device powers up.

> As scheduler operation depends upon the availability of the updated current time provided by the Time Server,
> it is application's responsibility to call :cpp:func:`bt_mesh_scheduler_srv_time_update` after current local
> time has been updated to schedule available entries correctly.

Reviewer: ```suggestion
The scheduler operation depends on the availability of the updated current time provided by the Time Server.
It is the application's responsibility to call :c:func:`bt_mesh_scheduler_srv_time_update` after the current local
time has been updated to schedule available entries correctly.
```
The only thing I am not sure of here is if the scheduler should be capitalized or not... Do we mean scheduler in general, or the Scheduler model's operation? That would be the clarifying information.

Author: About scheduler capitalization: This is how it is in spec, so I tried to follow the same: Whenever scheduler is referred to as a general functionality of the model it is "scheduler", whenever it refers to the model name or register it is referred to as "Scheduler Server/Client" or "Scheduler Register state".

Reviewer: Ok, so it is meant as a general functionality here. Good, I agree, we leave it in lower case then :)
